src/Symfony/Component/Process/ExecutableFinder.php

```diff
@@ -48,9 +48,23 @@ public function addSuffix(string $suffix)
      */
     public function find(string $name, ?string $default = null, array $extraDirs = [])
     {
+        $dirs = array_merge(
+            explode(\PATH_SEPARATOR, getenv('PATH') ?: getenv('Path')),
+            $extraDirs
+        );
+
         if (\ini_get('open_basedir')) {
             $searchPath = array_merge(explode(\PATH_SEPARATOR, \ini_get('open_basedir')), $extraDirs);
-            $dirs = [];
+            foreach ($dirs as $index => $dir) {
+                foreach ($searchPath as $path) {
+                    if (str_starts_with($dir, $path)) {
+                        continue 2;
+                    }
+                }
+
+                unset($dirs[$index]);
+            }
```

Comment:
This isn't actually needed as 

```php
if (@is_file($file = $dir.\DIRECTORY_SEPARATOR.$name.$suffix) && ('\\' === \DIRECTORY_SEPARATOR || @is_executable($file))) {
                    return $file;
}
```

further below already only returns executables that are accessible by the PHP process (i.e. allowed by `open_basedir`).